frontend/packages/knative-plugin/src/actions/providers.ts

```diff
@@ -12,3 +20,40 @@ export const useMakeServerlessActionProvider = (resource: K8sResourceKind) => {
 
   return [deploymentActions, !inFlight, undefined];
 };
+
+export const useKnativeServiceActionsProvider = (resource: K8sResourceKind) => {
+  const [kindObj, inFlight] = useK8sModel(referenceFor(resource));
+  const actions = React.useMemo(() => {
+    return [
+      setTrafficDistribution(kindObj, resource),
+      getHealthChecksAction(kindObj, resource),
+      editKnativeService(kindObj, resource),
+      DeploymentActionFactory.EditResourceLimits(kindObj, resource),
+      ...getCommonResourceActions(kindObj, resource),
+    ];
+  }, [kindObj, resource]);
+
+  return [actions, !inFlight, undefined];
+};
+
+export const useCommonActionsProvider = (resource: K8sResourceKind) => {
+  const [kindObj, inFlight] = useK8sModel(referenceFor(resource));
+  const actions = React.useMemo(() => getCommonResourceActions(kindObj, resource), [
+    kindObj,
+    resource,
+  ]);
+  return [actions, !inFlight, undefined];
+};
+
+export const useKnativeEditApplicationActionProvider = (element: GraphElement) => {
+  const actions = React.useMemo(() => {
+    const resource = getResource(element);
+    const k8sKind = modelFor(referenceFor(resource));
+    return [getModifyApplicationAction(k8sKind, resource)];
+  }, [element]);
+
+  return React.useMemo(() => {
+    if (element.getType() !== NodeType.KnService) return [[], true, undefined];
```

Comment:
move this condition in the first use memo, otherwise UI will break for non-resource backed sidepanels